### PR TITLE
Switch to C# 8

### PR DESCRIPTION
**Pull request by DavidKarlas**

### CODE REVIEW

**Review by vlada-shubina — APPROVED** [review]

**Comment by vlada-shubina** [comment]
@DavidKarlas should we also go and change TFM to netcoreapp3.1 for those projects which are still on 2.1 now? 
- Microsoft.TemplateEngine.CLI
- Microsoft.TemplateSearch.*
- test projects (VS is not using our test frameworks afaik)

**Comment by DavidKarlas** [comment]
Notice that beside bumping some projects that were still on netcoreapp2.1(ones Vlada mentioned).
I also bumped net45 to net472. Visual Studio is using net472 too, and reason we want to target net472, is so we can use tuples...

**Review by vlada-shubina — COMMENTED** [review]
Test comment

**Comment by DavidKarlas** [comment]
@vlada-shubina since this is only effecting master... should we just go `net5.0`?

**Comment by vlada-shubina** [comment]
> @vlada-shubina since this is only effecting master... should we just go `net5.0`?

Let's go for it, but let's keep .NET framework aside for now until VS confirms we can get rid of it.

**Comment by ladipro** [comment]
FWIW, MSBuild is targeting `net472` and is being used in/with VS as well. It should be a safe change.

File: test/Microsoft.TemplateEngine.Cli.UnitTests/EndToEndTestBase.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public void Run(string args, params string[] scripts)
 #else
             string configuration = "Release";
 #endif
-            string harnessPath = Path.Combine(dir, "..", "..", "..", "..", "bin", "Microsoft.TemplateEngine.EndToEndTestHarness", configuration, "netcoreapp3.1");
+            string harnessPath = Path.Combine(dir, "..", "..", "..", "..", "bin", "Microsoft.TemplateEngine.EndToEndTestHarness", configuration, Path.GetFileName(dir));
```
Comment (donJoseLuis) [inline test/Microsoft.TemplateEngine.Cli.UnitTests/EndToEndTestBase.cs:25]: unsure if I understand this one 100%.  Can we either talk about it briefly or get someone else to approve this particular item?

**Review by donJoseLuis — APPROVED** [review]

File: test/Microsoft.TemplateEngine.Cli.UnitTests/EndToEndTestBase.cs (hunk 1 above)
Comment (DavidKarlas) [reply test/Microsoft.TemplateEngine.Cli.UnitTests/EndToEndTestBase.cs]: Made few changes to make it clearer what intention of that change is

**Pull request merged**